### PR TITLE
Switch from Maybe to Loaded for partySiteAccess field

### PR DESCRIPTION
**Pull request by kanishka-azimi**
## Description

In response to PR feedback, start implementing a more explicit type to denote fields that are not always loaded.

## Motivation and Context

Loaded type makes the implementation more self-documenting.

## How Has This Been Tested?

yes

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have run the unit tests.
- [x] I am sick of writing content for this PR, let's get this over with.

### CODE REVIEW

**Comment by chreekat** [comment]
Previously, I wasn't sure if you meant the *Party* was/wasn't loaded, or just the SiteAccess field within the Party. Now, I'm assuming it's the latter. But, what does it mean for a SiteAccess to be "loaded"?

**Comment by kanishka-azimi** [comment]
> But, what does it mean for a SiteAccess to be "loaded"?

I don't know how to word this any differently. Some queries join the data in and load that field, some don't.

File: src/Model/Party.hs [hunk 1]
```diff
@@ -1076,7 +1076,7 @@ findParties pf = do
                     vurl_a6R3W)
                  (do { cm_a6R44 <- vemail_a6R3X;
                        Just (Account cm_a6R44) })
-                 site)
+                 (maybe NotLoaded Loaded site))
```
Comment (chreekat) [inline src/Model/Party.hs]: >  Some queries join the data in and load that field, some don't.

Ahh, I see. in that case, though, isn't the value `Loaded` in both cases here? The data is joined and looked for, after all. If it still doesn't exist, that suggests a `Loaded (Maybe a)`

**Comment by chreekat** [comment]
ping, in case clicking 'approve changes' didn't already send a ping :)

**Pull request merged**